lua/wikis/ageofempires/Infobox/Faction/Custom.lua

```diff
@@ -0,0 +1,172 @@
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Array = require('Module:Array')
+local Class = require('Module:Class')
+local Game = require('Module:Game')
+local Lua = require('Module:Lua')
+local Page = require('Module:Page')
+
+local Injector = Lua.import('Module:Widget/Injector')
+local FactionInfobox = Lua.import('Module:Infobox/Faction')
+
+local Widgets = require('Module:Widget/All')
+local WidgetsHtml = require('Module:Widget/Html/All')
+local Cell = Widgets.Cell
+local Fragment = WidgetsHtml.Fragment
+local Image = require('Module:Widget/Image/Icon/Image')
+
+---@class AoECustomFactionInfobox: FactionInfobox
+---@field game string
+local CustomFactionInfobox = Class.new(FactionInfobox)
+local CustomInjector = Class.new(Injector)
+
+---@param frame Frame
+---@return string
+function CustomFactionInfobox.run(frame)
+	local infobox = CustomFactionInfobox(frame)
+
+	infobox.args.informationType = 'Civilization'
+	infobox.args.lpdbType = 'civ'
+
+	local subpageText = mw.title.getCurrentTitle().subpageText
+	infobox.game = Game.toIdentifier{
+		game = infobox.args.game or subpageText
+	}
+
+	if infobox.game == subpageText then
+		infobox.name = mw.title.getCurrentTitle().baseText
+	end
+
+	infobox:setWidgetInjector(CustomInjector(infobox))
+
+	return infobox:createInfobox()
+end
+
+function CustomInjector:parse(id, widgets)
+	---@type AoECustomFactionInfobox
+	local caller = self.caller
+	local args = caller.args
+
+	if id == 'release' then
+		return {
+			args.introduced and Cell{
+				name = 'First introduced',
+				children = {
+					Fragment{
+						children = {
+							caller:_makeIntroducedIcon(args.introduced),
+							Page.makeInternalLink(args.introduced)
+						}
+					}
+				}
+			} or Fragment{}
+		}
+	elseif id == 'custom' then
+		Array.appendWith(widgets,
+			Cell{
+				name = Page.makeInternalLink('Architectural Style', 'Architectures (building styles)'),
+				children = {args.architecture}
+			},
+			Cell{name = 'Continent', children = {args.continent}},
+			Cell{
+				name = Page.makeInternalLink('Ingame classification', 'Civilizations classification'),
+				children = Array.map(
+					caller:getAllArgsForBase(args, 'type'),
+					function (t)
+						return t .. ' civilization'
+					end
+				)
+			},
+			Cell{
+				name = 'Unique buildings',
+				children = Array.map(
+					caller:getAllArgsForBase(args, 'building'),
+					Page.makeInternalLink
+				)
+			},
+			Cell{
+				name = 'Unique units',
+				children = Array.map(
+					caller:getAllArgsForBase(args, 'unit'),
+					Page.makeInternalLink
+				)
+			},
+			args.tech1 and Cell{
+				name = 'Unique technologies',
+				children = {
+					Fragment{
+						children = {
+							caller:_makeAgeIcon('Castle'),
+							Page.makeInternalLink(args.tech1)
+						}
+					},
+					Fragment{
+						children = {
+							caller:_makeAgeIcon('Imperial'),
+							Page.makeInternalLink(args.tech2)
+						}
+					}
+				}
```

Comment:
```suggestion
				children = {
					caller:_makeAgeIcon('Castle'),
					Page.makeInternalLink(args.tech1)
					caller:_makeAgeIcon('Imperial'),
					Page.makeInternalLink(args.tech2)
				}
```
I think the resulting html in both cases are the same.

Are you sure this is what you're intending?